include/grpc++/impl/codegen/method_handler_impl.h

```diff
@@ -27,6 +27,25 @@
 namespace grpc {
 
 namespace internal {
+
+// Invoke the method handler, fill in the status, and
+// return whether or not we finished safely (without an exception).
+// Note that exception handling is 0-cost in most compiler/library
+// implementations (except when an exception is actually thrown),
+// so this process doesn't require additional overhead in the common case.
+// Additionally, we don't need to return if we caught an exception or not;
+// the handling is the same in either case.
+template <class F>
+Status CatchingFunctionHandler(F&& callable) {
+  try {
+    return callable();
```

Comment:
It's probably worth providing a preprocessor control for this: there exist codebases where the existence of try/catch will be a deal breaker

Reply:
Sure, I was starting to question whether this should be considered an API breaker since it no longer allows compiling with "-fno-exceptions"

Reply:
Ok, I put in the guards